chain/jsonrpc-primitives/src/types/transactions.rs

```diff
@@ -0,0 +1,152 @@
+        {
+            if !near_runtime_utils::is_valid_account_id(&account_id) {
+                return Err(crate::errors::RpcParseError(format!(
+                    "Invalid account id: {}",
+                    account_id
+                )));
+            }
+            let transaction_info = TransactionInfo::TransactionId { hash, account_id };
+            Ok(Self { transaction_info })
+        } else {
+            // TODO: move this code to crate::utils::parse_signed_transaction
+            let (encoded,) = crate::utils::parse_params::<(String,)>(value.clone())?;
+            let bytes = crate::utils::from_base64_or_parse_err(encoded)?;
+            let signed_transaction =
+                near_primitives::transaction::SignedTransaction::try_from_slice(&bytes).map_err(
+                    |err| {
+                        crate::errors::RpcParseError(format!(
+                            "Failed to decode transaction: {}",
+                            err
+                        ))
+                    },
+                )?;
+            let transaction_info = TransactionInfo::Transaction(signed_transaction);
+            Ok(Self { transaction_info })
+        }
+    }
+}
+
+impl From<near_client_primitives::types::TxStatusError> for RpcTransactionError {
+    fn from(error: near_client_primitives::types::TxStatusError) -> Self {
+        match error {
+            near_client_primitives::types::TxStatusError::ChainError(err) => {
+                Self::InternalError { debug_info: format!("{:?}", err) }
+            }
+            near_client_primitives::types::TxStatusError::MissingTransaction(
+                requested_transaction_hash,
+            ) => Self::UnknownTransaction { requested_transaction_hash },
+            near_client_primitives::types::TxStatusError::InvalidTx(context) => {
+                Self::InvalidTransaction { context }
+            }
+            near_client_primitives::types::TxStatusError::InternalError => {
+                Self::InternalError { debug_info: format!("Internal error") }
+            }
+            near_client_primitives::types::TxStatusError::TimeoutError => Self::TimeoutError,
+        }
+    }
+}
+
+impl From<near_primitives::views::FinalExecutionOutcomeViewEnum> for RpcTransactionResponse {
+    fn from(
+        final_execution_outcome: near_primitives::views::FinalExecutionOutcomeViewEnum,
+    ) -> Self {
+        Self { final_execution_outcome }
+    }
+}
+
+impl From<RpcTransactionError> for crate::errors::RpcError {
+    fn from(error: RpcTransactionError) -> Self {
+        let error_data = match error {
+            RpcTransactionError::InvalidTransaction { ref context } => Some(
+                serde_json::to_value(crate::errors::ServerError::TxExecutionError(
+                    near_primitives::errors::TxExecutionError::InvalidTxError(context.clone()),
+                ))
+                .unwrap_or(Value::String(error.to_string())),
```

Comment:
Looks like there's some leftover code here?


```suggestion
            RpcTransactionError::InvalidTransaction { ref context } =>
                serde_json::to_value(crate::errors::ServerError::TxExecutionError(
                    near_primitives::errors::TxExecutionError::InvalidTxError(context.clone()),
                )
```

Or should this be `.ok()` rather then `Some()`?